gamemodes/horde/entities/weapons/arccw_horde_medic_shotgun.lua

```diff
@@ -0,0 +1,390 @@
+        ShellEjectAt = 0,
+    },
+    ["cycle_iron"] = {
+        Source = "shoot",
+        Time = 0.7,
+        ShellEjectAt = 0,
+    },
+    ["sgreload_start"] = {
+        Source = "start_reload",
+        Time = 0.2,
+        TPAnim = ACT_HL2MP_GESTURE_RELOAD_SHOTGUN,
+        LHIK = true,
+        LHIKIn = 0.5,
+        LHIKOut = 0,
+    },
+    ["sgreload_insert"] = {
+        Source = "insert",
+        Time = 0.5,
+        TPAnim = ACT_HL2MP_GESTURE_RELOAD_SHOTGUN,
+        TPAnimStartTime = 0.3,
+        LHIK = true,
+        LHIKIn = 0,
+        LHIKOut = 0,
+    },
+    ["sgreload_finish"] = {
+        Source = "reload_end",
+        Time = 1,
+        LHIK = true,
+        LHIKIn = 0,
+        LHIKOut = 0.4,
+    },
+    ["sgreload_finish_empty"] = {
+        Source = "reload_end",
+        Time = 1,
+        LHIK = true,
+        LHIKIn = 0,
+        LHIKOut = 0.4,
+    },
+}
+
+function SWEP:DoPrimaryFire(isent, data)
+    local clip = self:Clip1()
+    if self:HasBottomlessClip() then
+        clip = self:Ammo1()
+    end
+    local owner = self:GetOwner()
+
+    local shouldphysical = GetConVar("arccw_bullet_enable"):GetBool()
+
+    if self.AlwaysPhysBullet or self:GetBuff_Override("Override_AlwaysPhysBullet") then
+        shouldphysical = true
+    end
+
+    if self.NeverPhysBullet or self:GetBuff_Override("Override_NeverPhysBullet") then
+        shouldphysical = false
+    end
+
+    if isent then
+        self:FireRocket(data.ent, data.vel, data.ang, self.PhysBulletDontInheritPlayerVelocity)
+    else
+        -- if !game.SinglePlayer() and !IsFirstTimePredicted() then return end
+        if !IsFirstTimePredicted() then return end
+
+        if shouldphysical then
+            local vel = self:GetBuff_Override("Override_PhysBulletMuzzleVelocity") or self.PhysBulletMuzzleVelocity
+
+            local tracernum = data.TracerNum or 1
+            local prof
+
+            if tracernum == 0 or clip % tracernum != 0 then
+                prof = 7
+            end
+
+            if !vel then
+                vel = math.Clamp(self:GetBuff("Range"), 30, 300) * 8
+
+                if self.DamageMin > self.Damage then
+                    vel = vel * 3
+                end
+            end
+
+            vel = vel / ArcCW.HUToM
+
+            vel = vel * self:GetBuff_Mult("Mult_PhysBulletMuzzleVelocity") * self:GetBuff_Mult("Mult_Range")
+
+            vel = vel * GetConVar("arccw_bullet_velocity"):GetFloat()
+
+            vel = vel * data.Dir:GetNormalized()
+
+            ArcCW:ShootPhysBullet(self, data.Src, vel, prof)
+        else
+            if owner:IsPlayer() then
+                owner:LagCompensation(true)
+            end
+            self:FireBullets(data)
+            if owner:IsPlayer() then
+                owner:LagCompensation(false)
+            end
+        end
+    end
+end
+
+hook.Add("Hook_BulletHit", "Horde_MedicShotgunHeal", function(wpn, data)
+    if wpn:GetClass() == "arccw_horde_medic_shotgun" then
```

Comment:
This should return early <https://github.com/CFC-Servers/cfc_glua_style_guidelines/#return-early-from-functions> rather than being a big if statement